File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        l = 0
        maxi = 0
        for p in range(len(s)):
            while s[p] in seen:
                seen.remove(s[l])
                l+=1
            seen.add(s[p])
            maxi = max(maxi, p-l+1)
        return maxi

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    slice := make(map[byte]bool)
    maxi := 0
    l := 0
    for i:= 0; i <len(s); i++{
        for slice[s[i]]{
            delete(slice, s[l])
            l++
        }
        slice[s[i]] = true
        maxi = max(maxi, i - l +1)
    }
    return maxi
}